Filter/Basic.php

```diff
@@ -203,13 +203,13 @@ public function getParameters()
 
     /**
      * Get the stream resource being filtered.
-     * Maybe available only during filter calls when the closing parameter is
-     * set to false.
+     * Maybe available only during `static::filter` calls when the closing
+     * parameter is set to false.
      *
      * @return  resource
      */
     public function getStream()
     {
-        return $this->stream;
+        return isset($this->stream) ? $this->stream : null;
```

Comment:
Yup, go check this demo from AFUP Lyon 2013 event: http://keynote.hoa-project.net/AFUPLyon13/Demonstration/03_Stream_filter.php.